feat: Improve warm cody web startup time

fkling opened the pull request:
One major problem with cody web right now is that we recreate the cody agent whenever we navigate to a page that shows cody web components.
Worse even, on e.g. the prompts page where we should three prompt inputs, we create three separate instances of the agent.

This initialization step takes time and uses server resources (having already caused Cloudflare overages (https://sourcegraph.slack.com/archives/C05EA9KQUTA/p1737688274086469)).

This PR proposes to only use a single instance of cody agent.

> [!IMPORTANT]
> My understanding of cody web and cody agent is limited. I don't know what implication it has to 'reuse' the same cody agent instance. I've manually tested the typical workflows (chatting, creating new chat, switching files with cody sidebar open, etc), but there are likely things I missed or am not aware of.

> [!IMPORTANT]
> This does **not** improve the initial loading time of cody web.

https://github.com/user-attachments/assets/dc5ed22f-15c2-4084-8c2e-a9cc133c38d9



## Test plan

Manual testing.

fkling commented:
This change is part of the following stack:

- #6947 ◀

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>
<!-- gs:navigation comment -->

dominiccooney reviewed (approved):
Awesome, I love it.

I'm interested in how shutdown will work, will go read the other PR.

Obviously this increases the possibility of leaks and so on, but it's a question of degree not kind and we should be fixing those anyway.

fkling commented:
@vovakulikov, @dominiccooney I've changed the implementation to move the agent initialization back into cody web, to keep things encapsulated. We still get the benefit of reusing a single agent instance.
I've also added logic to shutdown the agent if it hasn't been used for a while.

dominiccooney commented on web/lib/agent/agent.client.ts:
```diff
@@ -70,9 +68,7 @@ export async function createAgentClient({
-        // Empty root URI leads to openctx configuration resolution failure, any non-empty
-        // mock value (Cody Web doesn't really use any workspace related features)
-        workspaceRootUri: `repo:${repository ?? ''}`,
+        workspaceRootUri: '',
```
JetBrains populates that field for the initial workspace here: https://sourcegraph.sourcegraph.com/github.com/sourcegraph/cody/-/blob/jetbrains/src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt?L137-139

Cody for JetBrains does not send workspaceFolder/didChange.

JetBrains does have multi-root workspaces, but I believe the feature is pretty obscure.
